src/meta/processors/admin/HBProcessor.cpp

```diff
@@ -32,7 +32,7 @@ void HBProcessor::process(const cpp2::HBReq& req) {
   auto role = req.get_role();
   LOG(INFO) << "Receive heartbeat from " << host
             << ", role = " << apache::thrift::util::enumNameSafe(role);
-
+  folly::SharedMutex::WriteHolder holder(LockUtils::lock());
```

Comment:
Since heartbeat's num is way more than other meta processor, do we really have to add write lock here? What we need to write here is only related to one host.

Reply:
Mainly to prevent inconsistencies when reading data, this writing method is more strict.